dist_tables/dml.rst

```diff
@@ -38,6 +38,70 @@ For example:
+
+.. code-block:: postgres
+
+  INSERT INTO daily_page_views (day, page_id, view_count)
+    SELECT view_time::date AS day, page_id, count(*) AS view_count
+    FROM page_views
+    WHERE view_time >= $1
+    GROUP BY view_time::date, page_id
+    ON CONFLICT (day, page_id) DO UPDATE SET
+      view_count = daily_page_views.view_count + EXCLUDED.view_count;
+
+These are the basics of using insert into select on a single node database. Using it effectively in Citus for real-time analytics requires more understanding of Citus' distributed query execution. Citus implements insert into select by pushing down the select query to each shard. When the destination table for inserted values is distributed by an analogous column as the table being selected then the inserting and selecting can be "co-located" together on each shard. This minimizes network traffic between nodes and allows highly parallel execution.
+
+For fast execution ensure that:
+
+- The tables queried and inserted are distributed on analogous columns
+- The select query includes the distribution column
+- The insert statement includes the distribution column value
+- GROUP BY includes the distribution column
+
+The example query shown above will work well if :code:`page_views` and :code:`daily_page_views` are distributed by :code:`page_id` column. Distributing our data by page allows rollup to scale linearly with the number of cores.
+
+.. code-block:: postgres
+
+  SELECT create_distributed_table('page_views', 'page_id');
+  SELECT create_distributed_table('daily_page_views', 'page_id');
+
+There are a few things to be aware of when running distributed insert into selects. First, Citus takes locks to ensure consistency when the :ref:`replication_factor` is greater than one. In this situation if you know your inserts are not going to affect an ongoing rollup, you can set :code:`SET LOCAL citus.all_modifications_commutative TO on;`. This avoids strict locking and provides better throughput.
+
+Next bear in mind that any functions in the select statement for insertion must be immutable. Stable functions are not allowed, and volatile functions are certainly not allowed. The PostgreSQL docs explain the difference between these `volatility categories <https://www.postgresql.org/docs/current/static/xfunc-volatility.html>`_. Consider what would happen with replication factor greater than one. If you run :code:`now()` on the workers, the two replicas would differ. One surprising way that stable functions can sneak in is using timestamps with timezones (:code:`timestamptz`). Functions dealing with timezones rely implicitly on the timezone configuration parameter and hence are marked stable rather than immutable.
```

Comment:
Minor note -- this could be non-actionable.

When I read this paragraph, I'm a bit confused by the first paragraph. In particular, "any functions ... immutable" requires a certain knowledge of PostgreSQL functions and their properties. If we introduced function volatility, we could ease the user into "PG functions' properties."

We may not need to think about this though. We added support for stable functions in ``INSERT INTO SELECT`` in 6.1. Once we release 6.1, one option could be to remove this paragraph (#246)?